packages/xod-client/src/workers/reducer.js

```diff
@@ -0,0 +1,23 @@
+import * as R from 'ramda';
+import * as EAT from '../editor/actionTypes';
+
+const workersReducer = (state = {}, action) => {
+  switch (action.type) {
+    case EAT.TABTEST_LAUNCHED: {
+      return R.assoc(action.payload.workerId, action.payload.worker, state);
+    }
+    case EAT.TABTEST_RUN_FINISHED:
+    case EAT.TABTEST_ERROR: {
+      const workerId = action.meta.workerId;
+      const worker = R.prop(workerId, state);
+      if (worker) {
+        worker.terminate();
```

Comment:
Reducers must not have side effects. Try to make a pure test for this reducer? Termination should belong to a dispatcher or middleware.